opensciencegrid/contact-sync/topology_utils.py

```diff
@@ -0,0 +1,375 @@
+                        get_contact_list_info(contact_type))
+
+        if name and contact_list_info:
+            results[name] = contact_list_info
+
+    return results
+
+
+def get_resource_contacts_by_name_and_fqdn(args):
+    """
+    Get resource contacts for OSG.  Return results.
+
+    Returns two dictionaries, one keyed on the resource name and one keyed on
+    the resource FQDN.
+    """
+    root = get_contacts(args, 'rg', 'Resource')
+    if root is None:
+        return {}, {}
+
+    results_by_name = {}
+    results_by_fqdn = {}
+    for child_rg in root:
+        if child_rg.tag != "ResourceGroup":
+            print("MyOSG returned a non-resource group (%s) inside summary." % \
+                  root.tag, file=sys.stderr)
+            return {}, {}
+        for child_res in child_rg:
+            if child_res.tag != "Resources":
+                continue
+            for resource in child_res:
+                resource_name = None
+                resource_fqdn = None
+                contact_list_info = []
+                for resource_tag in resource:
+                    if resource_tag.tag == 'Name':
+                        resource_name = resource_tag.text
+                    if resource_tag.tag == 'FQDN':
+                        resource_fqdn = resource_tag.text
+                    if resource_tag.tag == 'ContactLists':
+                        for contact_list in resource_tag:
+                            if contact_list.tag == 'ContactList':
+                                contact_list_info.extend( \
+                                    get_contact_list_info(contact_list))
+
+                if contact_list_info:
+                    if resource_name:
+                        results_by_name[resource_name] = contact_list_info
+                    if resource_fqdn:
+                        results_by_fqdn[resource_fqdn] = contact_list_info
+    #print (results_by_name)
+    #print (results_by_fqdn)
+    return results_by_name, results_by_fqdn
+
+
+def get_resource_contacts(args):
+    return get_resource_contacts_by_name_and_fqdn(args)[0]
+
+
+def get_resource_contacts_by_fqdn(args):
+    return get_resource_contacts_by_name_and_fqdn(args)[1]
+
+#def filter_contacts(args, results):
+def filter_contacts(args, ori_results):
+    """
+    Given a set of result contacts, filter them according to given arguments
+    """
+    #results = dict(results)  # make a copy so we don't modify the original
+    results = dict(ori_results)
+    #print(results)
+
+    if getattr(args, 'name_filter', None):
+        # filter out undesired names
+        for name in ori_results.keys():
+        #for name in results.keys():
+            if not fnmatch.fnmatch(name, args.name_filter) and \
+                    args.name_filter not in name:
+                del results[name]
+    elif getattr(args, 'fqdn_filter', None):
+        # filter out undesired FQDNs
+        for fqdn in results.keys():
```

Comment:
```suggestion
        for fqdn in results:
```

again, `.keys()` is redundant here